src/python/pants/backend/project_info/register.py

```diff
@@ -23,7 +23,7 @@ def register_goals():
 
   task(name='depmap', action=Depmap).install()
   task(name='dependencies', action=Dependencies).install()
-  task(name='filedeps', action=FileDeps).install('filedeps')
+  task(name='legacy', action=FileDeps).install('filedeps')
```

Comment:
This leads to the weirdness where `./pants --v2 filedeps foo` will run both... It feels like we should atomically replace tasks, so if we're going to do this, we should either install the v2 version as the `experimental-filedeps` goal or the v1 version as the `legacy-filedeps` goal, rather than installing both under the same goal name...

Reply:
Well, you are implicitly running `--v1 --v2` when you do that, so I'm not sure that that is the wrong behaviour.

The intent behind having two flags might be invalidated by this change though. Will think on it some more.

Reply:
I've punted on this by renaming the `--v2` `filedeps` to `fast-filedeps` until it reaches parity.